code/game/objects/items/rcd/RPD.dm

```diff
@@ -528,50 +528,57 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 		switch(category) //if we've gotten this var, the target is valid
 			if(ATMOS_CATEGORY) //Making pipes
 				if(!can_make_pipe)
 					return
 				playsound(get_turf(src), 'sound/machines/click.ogg', 50, TRUE)
+				if(bluespace)
+					beam = user.Beam(attack_target, icon_state = "rped_upgrade", time = atmos_build_speed)
 				if (recipe.type == /datum/pipe_info/meter)
-					if(do_after(user, atmos_build_speed, target = attack_target))
-						playsound(get_turf(src), RPD_USE_SOUND, 50, TRUE)
-						var/obj/item/pipe_meter/new_meter = new /obj/item/pipe_meter(get_turf(attack_target))
-						new_meter.setAttachLayer(piping_layer)
-						if(mode & WRENCH_MODE)
-							new_meter.wrench_act(user, src)
+					if(!do_after(user, atmos_build_speed, target = attack_target))
+						if(!isnull(beam))
+							qdel(beam)
+					playsound(get_turf(src), RPD_USE_SOUND, 50, TRUE)
+					var/obj/item/pipe_meter/new_meter = new /obj/item/pipe_meter(get_turf(attack_target))
+					new_meter.setAttachLayer(piping_layer)
+					if(mode & WRENCH_MODE)
+						new_meter.wrench_act(user, src)
 				else
 					if(recipe.all_layers == FALSE && (piping_layer == 1 || piping_layer == 5))
 						balloon_alert(user, "can't build on this layer!")
 						return
-					if(do_after(user, atmos_build_speed, target = attack_target))
-						if(recipe.all_layers == FALSE && (piping_layer == 1 || piping_layer == 5))//double check to stop cheaters (and to not waste time waiting for something that can't be placed)
-							balloon_alert(user, "can't build on this layer!")
-							return
-						playsound(get_turf(src), RPD_USE_SOUND, 50, TRUE)
-						if(bluespace)
-							user.Beam(attack_target, icon_state = "rped_upgrade", time = 1)
-						var/obj/machinery/atmospherics/path = queued_p_type
-						var/pipe_item_type = initial(path.construction_type) || /obj/item/pipe
-						var/obj/item/pipe/pipe_type = new pipe_item_type(
-							get_turf(attack_target),
-							queued_p_type,
-							queued_p_dir,
-							null,
-							GLOB.pipe_paint_colors[paint_color],
-							ispath(queued_p_type, /obj/machinery/atmospherics/pipe/smart) ? p_init_dir : null,
-						)
-						if(queued_p_flipped && istype(pipe_type, /obj/item/pipe/trinary/flippable))
-							var/obj/item/pipe/trinary/flippable/new_flippable_pipe = pipe_type
-							new_flippable_pipe.flipped = queued_p_flipped
-
-						pipe_type.update()
-						pipe_type.add_fingerprint(usr)
-						pipe_type.set_piping_layer(piping_layer)
-						if(ispath(queued_p_type, /obj/machinery/atmospherics) && !ispath(queued_p_type, /obj/machinery/atmospherics/pipe/color_adapter))
-							pipe_type.add_atom_colour(GLOB.pipe_paint_colors[paint_color], FIXED_COLOUR_PRIORITY)
-						if(mode & WRENCH_MODE)
-							pipe_type.wrench_act(user, src)
+					if(bluespace)
+						beam = user.Beam(attack_target, icon_state = "rped_upgrade", time = atmos_build_speed)
+					if(!do_after(user, atmos_build_speed, target = attack_target))
+						if(!isnull(beam))
+							qdel(beam)
```

Comment:
Ditto

Reply:
Actually i missed review for this in few other places. Make sure to check everywhere.

Reply:
Basically write it in this format. the return FALSE is what you should include

https://github.com/tgstation/tgstation/blob/12d341a1878ad12299bcc8f0e69d70e6b160799d/code/game/objects/items/rcd/RCD.dm#L249-L251

Sorry if i posted one line more in my previous comment this confusion wouldn't have happened